### PR TITLE
It 1496 shahram

### PR DESCRIPTION
**Pull request by shahramsobhani**
@jhoblitt I think it should look better now.

### CODE REVIEW

**Comment by csilva-cl** [comment]
Hi Shahram, is this for a new instance of greylog ?

**Comment by shahramsobhani** [comment]
Hi @csilva-cl , yes it is.

**Comment by csilva-cl** [comment]
ok, thanks but you can use our instance, also we are using k8s now.

File: hieradata/node/graylog-01.tu.lsst.org.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+classes:
+#  - "profile::default"
+  - "profile::it::po"  # temporary for testing
+  - "profile::it::graylog"
```
Comment (hreinking) [inline hieradata/node/graylog-01.tu.lsst.org.yaml:5]: after PR 194 gets merge, the IT folder won't exist

File: site/profile/manifests/it/graylog.pp [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  }
+class { 'graylog::repository':
+  version => '3.0'
+}
+#################################################################################################################
+# Requirements for graylog::server
+$ssl_config_dir = '/etc/graylog/server'
+
+#TODO fine a better way of resolve this dependency
+  file{ '/etc/graylog':
```
Comment (hreinking) [inline site/profile/manifests/it/graylog.pp:46]: this is not needed if you are defining graylog::server class

File: site/profile/manifests/it/graylog.pp [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    command => "openssl req -x509 -days ${certificate_duration} -nodes -newkey ${ssl_key_algorithm} \
+      -config ${ssl_config_dir}/${ssl_config_filename} -keyout ${ssl_config_dir}/${ssl_keyout_filename} \
+      -out ${ssl_config_dir}/${ssl_graylog_cert_filename}",
+    onlyif  => 'test ! -f /etc/graylog/server/graylog-certificate.pem',
+    require => [File["${ssl_config_dir}/${ssl_config_filename}"]]
+}
+
+# Convert PKCS5 into PKCS8 encrypted
+
+$ssl_pkcs8_encrypted_filename = 'pkcs8-encrypted.pem'
```
Comment (hreinking) [inline site/profile/manifests/it/graylog.pp:91]: The PR just merged from @jhoblitt includes the letsencrypt module, which would avoid the need of creating "by hand" the self-sign certificates and then convert them to pkcs8

File: site/profile/manifests/it/po.pp [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  $enhancers = [ 'tree', 'oddjob', 'oddjob-mkhomedir', 'adcli', 'wget', 'nmap',
+  'openldap-clients', 'policycoreutils-python', 'tcpdump', 'openssl', 'nc',
+  'git', 'openssl-devel', 'telnet', 'acpid', 'lvm2', 'bash-completion', 'sudo' ]
+
+package { $enhancers: }
+# Firewall rules
+################################################################################
+
+$lsst_firewall_default_zone = lookup('lsst_firewall_default_zone')
+
```
Comment (hreinking) [inline site/profile/manifests/it/po.pp:16]: This is best to be handle over hiera

File: site/profile/templates/it/graylog_ssl_cfg.epp [hunk 5]
```diff
@@ -0,0 +1,23 @@
+[req]
```
Comment (hreinking) [inline site/profile/templates/it/graylog_ssl_cfg.epp:1]: there are modules to generate self-sign certs; no need to provide all the parameters

**Pull request closed without merging**